bpf/aya-bpf/src/helpers.rs

```diff
@@ -815,6 +809,27 @@ impl<T> From<*mut T> for PrintkArg {
     }
 }
 
+/// Expands the given byte array to `DST_LEN`, right-padding it with zeros. If
+/// `DST_LEN` is smaller than `SRC_LEN`, the array is instead truncated.
+///
+/// This function serves as a helper for the [`bpf_printk!`] macro.
+#[doc(hidden)]
+pub const fn zero_pad_array<const SRC_LEN: usize, const DST_LEN: usize>(
+    src: [u8; SRC_LEN],
+) -> [u8; DST_LEN] {
```

Comment:
In case you are wondering why I went with a generic zero padding function instead of a "append_zero" function: a signature such as `const fn append_zero<const N: usize>(a: [u8; N]) -> [u8; N + 1]` will not compile on stable Rust, due to the arithmetic in the return value. It is however perfectly happy to do arithmetic on the call-site.